packages/shared-state-pirania/files/etc/shared-state/hooks/pirania/generate_vouchers

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/lua
+
+--! LibreMesh
+--! Copyright (C) 2019  Marcos Gutierrez <gmarcos@altermundi.net>
+--! Copyright (C) 2019  Luandro <luandro@gmail.com>
+--!
+--! This program is free software: you can redistribute it and/or modify
+--! it under the terms of the GNU Affero General Public License as
+--! published by the Free Software Foundation, either version 3 of the
+--! License, or (at your option) any later version.
+--!
+--! This program is distributed in the hope that it will be useful,
+--! but WITHOUT ANY WARRANTY; without even the implied warranty of
+--! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--! GNU Affero General Public License for more details.
+--!
+--! You should have received a copy of the GNU Affero General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+local JSON = require("luci.jsonc")
+local uci = require("uci")
+local uci_conf = uci.cursor()
+
+local inputTable = JSON.parse(io.stdin:read("*all"))
+local dbPath = uci_conf:get("pirania", "base_config", "db_path")
+local outputTable ={"key,voucher,expiretime,uploadlimit,downloadlimit,amountofmacsallowed,usedmacs,"}
+
+local function shell(command)
+    local handle = io.popen(command)
+    local result = handle:read("*a")
+    handle:close()
+    return result
+end
+
+function macs2string(macs)
```

Comment:
I meant that this whole function could be substituted by table.concat, It's me or the function is still there?

Reply:
done in 3fd69c9c4af326ac64da7e012748fb60a35c68a1